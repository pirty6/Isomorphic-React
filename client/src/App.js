import React, { Component } from 'react';
import './App.css';

import Page from './components/Page';

class App extends Component {
  // Initialize state
  render() {
    return (
      <div className="App">
        <Page/>
      </div>
    );
  }
}

export default App;
