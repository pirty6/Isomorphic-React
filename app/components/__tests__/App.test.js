import React from 'react';
import { shallow, mount } from 'enzyme';
import { StaticRouter, Redirect } from 'react-router-dom';

import App from '../App';

const mockPokemon = {
    list: [

    ]
}

describe('App', () => {
    it('should render Home component for route /', () => {
        const component = mount(
            <StaticRouter context={{}} location={'/'}>
                <App/>
            </StaticRouter>
        );
        expect(component).toMatchSnapshot() 
    })

    it('should render List component for route /pokemon/ability/testing', () => {
        const component = mount(
            <StaticRouter context={{}} location={'/pokemon/ability/testing'}>
                <App pokemon={mockPokemon} />
            </StaticRouter>
        );
        expect(component).toMatchSnapshot() 
    })

    it('should render Redirect component for route /pokemon', () => {
        const component = mount(
            <StaticRouter context={{}} location={'/pokemon'}>
                <App />
            </StaticRouter>
        );
        expect(component).toMatchSnapshot() 
        expect(component.find(Redirect).props().to).toEqual('/pokemon/ability/telepathy')
    })
})