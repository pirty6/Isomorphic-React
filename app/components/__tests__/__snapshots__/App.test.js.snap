// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render Home component for route / 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <div>
        Your SSR React Router Node App initialised with data!
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
              }
            }
            path="/"
          >
            <Home
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              staticContext={Object {}}
            >
              <div>
                 You are on your home page 
              </div>
            </Home>
          </Route>
        </Switch>
      </div>
    </App>
  </Router>
</StaticRouter>
`;

exports[`App should render List component for route /pokemon/ability/testing 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/pokemon/ability/testing"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/pokemon/ability/testing",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App
      pokemon={
        Object {
          "list": Array [],
        }
      }
    >
      <div>
        Your SSR React Router Node App initialised with data!
        <Switch>
          <Route
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {
                  "ability": "testing",
                },
                "path": "/pokemon/ability/:ability",
                "url": "/pokemon/ability/testing",
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/pokemon/ability/testing",
                "search": "",
              }
            }
            path="/pokemon/ability/:ability"
            render={[Function]}
          >
            <List
              location={
                Object {
                  "history": Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/pokemon/ability/testing",
                      "search": "",
                    },
                    "push": [Function],
                    "replace": [Function],
                  },
                  "location": Object {
                    "hash": "",
                    "pathname": "/pokemon/ability/testing",
                    "search": "",
                  },
                  "match": Object {
                    "isExact": true,
                    "params": Object {
                      "ability": "testing",
                    },
                    "path": "/pokemon/ability/:ability",
                    "url": "/pokemon/ability/testing",
                  },
                  "staticContext": Object {},
                }
              }
              pokemon={Array []}
            >
              <div>
                <h3>
                  testing
                </h3>
                <ul />
              </div>
            </List>
          </Route>
        </Switch>
      </div>
    </App>
  </Router>
</StaticRouter>
`;

exports[`App should render Redirect component for route /pokemon 1`] = `
<StaticRouter
  basename=""
  context={
    Object {
      "action": "REPLACE",
      "location": Object {
        "hash": "",
        "pathname": "/pokemon/ability/telepathy",
        "search": "",
      },
      "url": "/pokemon/ability/telepathy",
    }
  }
  location="/pokemon"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/pokemon",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <div>
        Your SSR React Router Node App initialised with data!
        <Switch>
          <Route
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/pokemon",
                "url": "/pokemon",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "pathname": "/pokemon",
                "search": "",
              }
            }
            path="/pokemon"
            render={[Function]}
          >
            <Redirect
              push={false}
              to="/pokemon/ability/telepathy"
            />
          </Route>
        </Switch>
      </div>
    </App>
  </Router>
</StaticRouter>
`;
